test/inject/testdata/injected_default.golden

```diff
@@ -11,6 +11,7 @@ spec:
   template:
     metadata:
       annotations:
+        config.linkerd.io/disable-identity: "true"
```

Comment:
for my own understanding, how does this interact with `linkerd.io/identity-mode: disabled` ?

Reply:
if we have `config.linkerd.io/disable-identity: "true"` (set manually by the user or by the system through `linkerd inject --disable-identity`) then the system will also set `linkerd.io/identity-mode: disabled`. 
Which is a little redundant but the reasoning is similar to what we had in #2593 for having both annotations for proxy-version, in that one is an explicit override present in a spec that might not have been injected yet, and the other is the actual config value used when the injection is performed.

Reply:
Ok thanks for the explanation. Naively this does feel a bit redundant. Something to keep in mind as we go along.